src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -44539,6 +44567,9 @@ static void F<T>()
                 // (6,9): warning CS8602: Dereference of a possibly null reference.
                 //         s.ToString();
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(6, 9),
+                // (7,17): warning CS8653: A default expression introduces a null value when 'T' is a non-nullable reference type.
```

Comment:
>// (7,17): warning CS8653: A default expression introduces a null value when 'T' is a non-nullable reference type. [](start = 16, length = 114)

Is this warning expected?

Reply:
I think so. The alternative would be to treat `default(T?)` as non-null value, which seems stranger.

---
In reply to: [314125833](https://github.com/dotnet/roslyn/pull/37596#discussion_r314125833) [](ancestors = 314125833)